systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -1450,6 +1457,157 @@ void testClientsCACertRenew() {
+
+        KafkaTopicResource.createAndWaitForReadiness(KafkaTopicResource.topic(clusterName, TOPIC_NAME).build());
+
+        LOGGER.info("Check KafkaUser certificate.");
+        KafkaUser user = KafkaUserResource.createAndWaitForReadiness(KafkaUserResource.tlsUser(clusterName, USER_NAME).build());
+        X509Certificate userCert = SecretUtils.getCertificateFromSecret(kubeClient().getSecret(USER_NAME), "user.crt");
+        assertThat("Generated ClientsCA does not have expected test Subject: " + userCert.getIssuerDN(),
+                SystemTestCertManager.containsAllDN(userCert.getIssuerX500Principal().getName(), STRIMZI_TEST_CLIENTS_CA));
+
+        LOGGER.info("Send and receive messages over TLS.");
+        KafkaClientsResource.createAndWaitForReadiness(KafkaClientsResource.deployKafkaClients(true, clusterName + "-" + Constants.KAFKA_CLIENTS, user).build());
+        final String kafkaClientsPodName =
+                ResourceManager.kubeClient().listPodsByPrefixInName(clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(TOPIC_NAME)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(clusterName)
+                .withKafkaUsername(USER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                .build();
+
+        LOGGER.info("Check for certificates used within kafka pod internal clients (producer/consumer)");
+        List<VolumeMount> volumeMounts = ResourceManager.kubeClient().listPodsByPrefixInName(clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getSpec().getContainers().get(0).getVolumeMounts();
+        for (VolumeMount vm : volumeMounts) {
+            if (vm.getMountPath().contains("user-secret-" + internalKafkaClient.getKafkaUsername())) {
+                assertThat("UserCert Issuer DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "issuer", STRIMZI_INTERMEDIATE_CA));
+                assertThat("UserCert Subject DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "subject", STRIMZI_TEST_CLIENTS_CA));
+
+            } else if (vm.getMountPath().contains("cluster-ca-" + internalKafkaClient.getKafkaUsername())) {
+                assertThat("ClusterCA Issuer DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "issuer", STRIMZI_INTERMEDIATE_CA));
+                assertThat("ClusterCA Subject DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "subject", STRIMZI_TEST_CLUSTER_CA));
+            }
+        }
```

Comment:
As we discussed, I don't think that this make sense here. You are just checking that the secrets are correctly mounted.
If you change the names of the certificates and then you'll use it for sending/receiving messages, you'll know that:

1) secrets are correctly mounted
2) certs are working